src/base/memory_monitor_sr.cpp

```diff
@@ -358,25 +421,49 @@ namespace cubperf
     return 0;
   }
 
-  int memory_monitor::add_stat (THREAD_ENTRY *thread_p, MMON_STAT_ID stat_id, uint64_t size, bool expand)
+  void memory_monitor::add_stat (MMON_STAT_ID stat_id, int64_t size)
   {
-    return 0;
+    int module_idx = get_module_idx (stat_id);
+
+    m_total_mem_usage += size;
+
+    m_module[module_idx]->add_stat (stat_id, size);
   }
 
-  int memory_monitor::sub_stat (THREAD_ENTRY *thread_p, MMON_STAT_ID stat_id, uint64_t size)
+  void memory_monitor::add_tran_stat (THREAD_ENTRY *thread_p, int64_t size)
   {
-    return 0;
+    LOG_TDES *tdes = LOG_FIND_CURRENT_TDES (thread_p);
+    assert (tdes != NULL);
```

Comment:
I think it can be removed since if it's NULL the right below line spit a segfault. Nothing undefined happens with the NULL tdes.